<?xml version="1.0" ?>
<launch>
  <arg name="robot" default="panda" doc="choose your robot. Possible values: [panda, fr3]"/>
  <arg name="arm_id" default="$(arg robot)" />
  <arg name="x_axis_dist" default="0.10" />
  <arg name="y_axis_dist" default="0.10" />
  <arg name="z_axis_dist" default="0.10" />

  <include file="$(find franka_control)/launch/franka_control.launch" pass_all_args="true"/>
  <rosparam command="load" file="$(find franka_example_controllers)/config/franka_example_controllers.yaml" subst_value="true" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  args="cartesian_pose_example_controller"/>
    <param name="x_axis_dist" value="$(arg x_axis_dist)" /> 
    <param name="y_axis_dist" value="$(arg y_axis_dist)" /> 
    <param name="z_axis_dist" value="$(arg z_axis_dist)" /> 
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_example_controllers)/launch/robot.rviz -f $(arg arm_id)_link0 --splash-screen $(find franka_visualization)/splash.png"/>
</launch>
